specification/securityinsights/resource-manager/Microsoft.SecurityInsights/preview/2022-04-01-preview/FileImports.json

```diff
@@ -0,0 +1,499 @@
+              "$ref": "../../../common/2.0/types.json#/definitions/CloudError"
+            }
+          }
+        }
+      },
+      "delete": {
+        "x-ms-examples": {
+          "Delete a file import.": {
+            "$ref": "./examples/fileImports/DeleteFileImport.json"
+          }
+        },
+        "tags": [
+          "FileImports"
+        ],
+        "description": "Delete the file import.",
+        "operationId": "FileImports_Delete",
+        "parameters": [
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "../../../common/2.0/types.json#/parameters/WorkspaceName"
+          },
+          {
+            "$ref": "#/parameters/FileImportId"
+          }
+        ],
+        "responses": {
+          "202": {
+            "description": "Accepted"
+          },
+          "204": {
+            "description": "No Content"
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "../../../common/2.0/types.json#/definitions/CloudError"
+            }
+          }
+        }
+      }
+    }
+  },
+  "definitions": {
+    "FileMetadata": {
+      "description": "Represents a file.",
+      "properties": {
+        "fileFormat": {
+          "description": "The format of the file",
+          "enum": [
+            "CSV",
+            "JSON",
+            "XML",
+            "Other"
+          ],
+          "type": "string",
+          "x-ms-enum": {
+            "modelAsString": true,
+            "name": "fileFormat",
+            "values": [
+              {
+                "description": "A CSV file.",
+                "value": "CSV"
+              },
+              {
+                "description": "A JSON file.",
+                "value": "JSON"
+              },
+              {
+                "description": "An XML file.",
+                "value": "XML"
+              },
+              {
+                "description": "A file of other format.",
+                "value": "Other"
+              }
+            ]
+          }
+        },
+        "fileName": {
+          "description": "The name of the file.",
+          "type": "string"
+        },
+        "fileSize": {
+          "description": "The size of the file.",
+          "format": "int32",
+          "type": "integer"
+        },
+        "fileContentUri": {
+          "description": "A URI with a valid SAS token to allow uploading / downloading the file.",
+          "type": "string"
+        },
+        "isDeleted": {
```

Comment:
Let's not use boolean. Please use enum instead. Something like fileStatus enum with value Deleted, Active...